otter/models/intents.py

```diff
@@ -0,0 +1,16 @@
+from characteristic import attributes
+
+from effect.twisted import deferred_performer
+
+
+@attributes(['scaling_group', 'modifier'])
+class ModifyGroupState(object):
```

Comment:
On first impression I'm not really keen on intents like that. My concern is that an intent like that would be _too_ flexible and encourage writing over-effectful code. It's on a continuum with very specific stuff like `ModifyGroupState` at one end, and `Func` on the other.

You could consider that intents should just be like effectful functions (or "procedures", in Pascal lingo). We shouldn't be afraid to write intents for the specific side-effecting operations we need to perform -- and on the other side of the coin, it would be kind of strange to have an actual procedure like `group_model_request` which took a method name and args and kwargs which we used any time we wanted to invoke a group operations.

However, I do understand that defining these intents can be tedious (not only needing a class and a function, but also a change to another file, effect_dispatcher.py). But if this is the main concern, instead of creating very general intents like `GroupModelRequest`, we could instead abstract the _definition_ of intents like these so that we can create them more efficiently -- for example, if we end up creating a bunch of intents that map 1:1 with ScalingGroup methods, imagine that we could actually have an `otter/models/intents.py` that looks like this:

```
performers = {}
def make_group_intent(method_name, arguments):
    """Create an intent class and performer, and register the performer in performers"""
    ...

ModifyGroupState = make_group_intent('modify_state', ['modifier'])
ViewGroupConfig = make_group_intent('view_config', [])

dispatcher = TypeDispatcher(performers)
```

Then, in effect_dispatcher.py, we could just compose in this `otter.models.intents.dispatcher` so we wouldn't need to update that file when we add a new intent here.

Anyway, just a thought exercise.